torch/_dynamo/symbolic_convert.py

```diff
@@ -581,6 +581,33 @@ def handle_graph_break(
     return decorator
 
 
+def promote_exception_to_graph_break(inner_fn):
+    @functools.wraps(inner_fn)
+    def wrapper(self: "InstructionTranslatorBase", inst: Instruction):
+        try:
+            TracingContext.set_current_loc(
+                self.f_code.co_filename, self.lineno, self.f_code.co_name
+            )
+            return inner_fn(self, inst)
+        except Unsupported:
+            raise
+        except RuntimeError as e:
+            if inst.offset is None or not hasattr(self.f_code, "co_exceptiontable"):
+                raise
+
+            from torch._dynamo.bytecode_transformation import parse_exception_table
+
+            table = parse_exception_table(self.f_code.co_exceptiontable)
+            for entry in table:
+                if entry.start <= inst.offset < entry.end:
+                    # If we are in a try block of a try-catch, promote the
+                    # exception to graph break
+                    raise Unsupported(str(e)) from e
+            raise
```

Comment:
Consider pulling this code into a helper function. I can imagine other code wanting to know if a given instr is within an exception handling block.

Reply:
Based on a simple code search, this is the only place we currently do this. Do you think it is valueable create the helper function now or if anything else needs it to do it at that time?

Reply:
IMO yes - it encapsulates the functionality and also provides immediate documentation on intent for code readers (`self.is_instr_in_exception_handler(instr)` is a lot more readable what the code is doing...). But up to you - it wouldn't hold me up from accepting.

Reply:
Done